internal/xds/server.go

```diff
@@ -68,3 +69,39 @@ func RegisterServer(srv Server, registry *prometheus.Registry, opts ...grpc.Serv
 
 	return g
 }
+
+// RegisterEnvoyServer registers the given xDS protocol Server with the gRPC
+// runtime. If registry is non-nil gRPC server metrics will be automatically
+// configured and enabled.
+func RegisterEnvoyServer(registry *prometheus.Registry, sc cache.SnapshotCache, opts ...grpc.ServerOption) *grpc.Server {
```

Comment:
It looks like you can probably de-dupe a lot of the implementation of this function with `RegisterServer` - seems like the only difference is that this one is instantiating the `xdsServer` directly rather than having it be passed in. You could either extract a helper that's used internally within this package, or you could instantiate the go-control-plane xds server in `serve.go` directly like the contour one, and then I don't think you even need `RegisterEnvoyServer`, you could just reuse `RegisterServer`.

Reply:
sweet, I was able to do that and remove the extra RegisterEnvoyServer method.